From a5f05ba5c03cd8b6d1e384b25a28013619329b48 Mon Sep 17 00:00:00 2001
From: Piotr Karbowski <piotr.karbowski@gmail.com>
Date: Fri, 21 Dec 2012 15:54:07 +0100
Subject: [PATCH] umask(077) as a workaround for security vuln.

Mdev seems to alter permissions of created dirs. Example:
microcode		root:root 600 =cpu/
cpu([0-9]+)		root:root 600 =cpu/%1/cpuid
msr([0-9]+)		root:root 600 =cpu/%1/msr

will make /dev/cpu a world-writtable dir.
---
 util-linux/mdev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util-linux/mdev.c b/util-linux/mdev.c
index 79871d3..b85e2d7 100644
--- a/util-linux/mdev.c
+++ b/util-linux/mdev.c
@@ -823,7 +823,7 @@ int mdev_main(int argc UNUSED_PARAM, char **argv)
 	bb_sanitize_stdio();
 
 	/* Force the configuration file settings exactly */
-	umask(0);
+	umask(077);
 
 	xchdir("/dev");
 
-- 
1.8.0.2

